In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
Above code is importing necessary modules for the exercise.

In [22]:
gdp_df = pd.read_csv("..\data\gdp_percapita.csv")
gdp_df

Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2020  1970.560169              NaN
1        Afghanistan  2019  2065.036235              NaN
2        Afghanistan  2018  2033.804389              NaN
3        Afghanistan  2017  2058.400221              NaN
4        Afghanistan  2016  2057.067978              NaN
...              ...   ...          ...              ...
7171        Zimbabwe  1994  2603.927369              NaN
7172        Zimbabwe  1993  2420.092791              NaN
7173        Zimbabwe  1992  2437.152908              NaN
7174        Zimbabwe  1991  2733.696626              NaN
7175        Zimbabwe  1990  2652.129209              NaN

[7176 rows x 4 columns]

Above code is importing the appropriate .csv where '..' symbolizes the need to move up in the file structure to search for the needed file. 

Code below is finding the layout of the DataFrame where the first number represents number of rows and the second represents number of columns. 

In [19]:
gdp_df.shape

(7176, 4)

#### How many rows and columns are in `gdp_df`? What are the data types of each column?

7176 rows and 4 columns.

In [28]:
gdp_df = gdp_df.drop(columns = ["Value Footnotes"])

Above code is dropping the "Value Footnotes" column.

In [29]:
gdp_df.columns = ['Country', 'Year', 'GDP_Per_capita']

In [30]:
gdp_df

Country  Year  GDP_Per_capita
0     Afghanistan  2020     1970.560169
1     Afghanistan  2019     2065.036235
2     Afghanistan  2018     2033.804389
3     Afghanistan  2017     2058.400221
4     Afghanistan  2016     2057.067978
...           ...   ...             ...
7171     Zimbabwe  1994     2603.927369
7172     Zimbabwe  1993     2420.092791
7173     Zimbabwe  1992     2437.152908
7174     Zimbabwe  1991     2733.696626
7175     Zimbabwe  1990     2652.129209

[7176 rows x 3 columns]

Above code is renaming the columns and displaying the new DataFrame

In [45]:
years = set(list(gdp_df.Year))
years

{1990,
 1991,
 1992,
 1993,
 1994,
 1995,
 1996,
 1997,
 1998,
 1999,
 2000,
 2001,
 2002,
 2003,
 2004,
 2005,
 2006,
 2007,
 2008,
 2009,
 2010,
 2011,
 2012,
 2013,
 2014,
 2015,
 2016,
 2017,
 2018,
 2019,
 2020}

Above code creates a set of all unique years in the DataFrame.